src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -7443,14 +7495,70 @@ instruct decI_mem(memory dst, immI_M1 src, rFlagsReg cr)
   ins_pipe(ialu_mem_imm);
 %}
 
-instruct leaI_rReg_immI(rRegI dst, rRegI src0, immI src1)
```

Comment:
I'm not sure if there would be any unexpected side-effect caused by the removal of `leaI_rReg_immI`.
Why you said it seems never be matched?

Reply:
The rule happens after `addI_rReg_imm` and has higher cost. I perceive it is there to define the peephole rule down below

https://github.com/openjdk/jdk/blob/b6c35ae44aeb31deb7a15ee2939156ed00b3df52/src/hotspot/cpu/x86/x86_64.ad#L13284